taichi/ir/expr.cpp

```diff
@@ -84,6 +84,10 @@ Expr &Expr::operator=(const Expr &o) {
   } else {
     set(o);  // Literally set this Expr to o
   }
+}
+
+Expr &Expr::operator=(const Expr &o) {
```

Comment:
Any chance we can remove this completely :-)

Reply:
I'm afraid not? When mark this as `=delete`, we can't compile taichi.

https://github.com/taichi-dev/taichi/blob/master/taichi/ir/expression.h#L89

```
9: error: overload resolution selected deleted operator '='
      *__position = std::forward<_Arg>(__arg);
      ~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/vector.tcc:154:6: note: in instantiation of function template specialization 'std::vector<taichi::lang::Expr, std::allocator<taichi::lang::Expr> >::_M_insert_aux<taichi::lang::Expr>' requested here
            _M_insert_aux(__pos, std::move(__x_copy._M_val()));
            ^
/home/jun/dev/cpp-projects/taichi/taichi/ir/expression.h:89:11: note: in instantiation of member function 'std::vector<taichi::lang::Expr, std::allocator<taichi::lang::Expr> >::insert' requested here
    exprs.insert(exprs.begin(), a);
          ^
/home/jun/dev/cpp-projects/taichi/taichi/ir/expr.h:78:9: note: candidate function has been explicitly deleted
  Expr &operator=(const Expr &o) = delete;
```
STL isn't happy about it...

Reply:
Oh sorry, I meant somerthing like `operator=(...) = default` (I.e. The thing that annoys me is the fact that `operator=` doesn't follow its canonical meaning..)

Reply:
I will try to work it out :-)

Reply:
> Oh sorry, I meant somerthing like `operator=(...) = default` (I.e. It's fine to keep the `operator=`. The thing that annoys me is the fact that `operator=` doesn't follow its canonical meaning...)

Got it!